src/main/java/com/lorepo/icplayer/client/module/choice/ChoicePrintable.java

```diff
@@ -21,49 +22,117 @@ public ChoicePrintable(ChoiceModel model, PrintableController controller) {
 	
 	public String getPrintableHTML(String className, boolean showAnswers) {
 		if (model.getPrintable() == PrintableMode.NO) return null;
-		
-		ArrayList<ChoiceOption> orderedOptions = new ArrayList<ChoiceOption>();
-		for (int i = 0; i < model.getOptionCount(); i++) {
-			orderedOptions.add(model.getOption(i));
+
+		HashMap<String, String> printableState = model.getPrintableState();
+		List<String> optionsHTMLs;

```

Comment:
To wygląda jak kontener niższego poziomu. Options wydaje się jako coś generycznego do reużycia.

Warto byłoby to opakować w swój własny kontener. Wtedy można byłoby przenieść tam takie funkcje fabryczne jak posortowanie opcji, wyciąganie opcji, czy tworzenie opcji na podstawie printable stanu a shows answers - zamiast zasmiecac kod w tym miejscu

Reply:
Została zrobiona klasa odpowiedzialna za wykonywanie operacji w zależności od stanu.